internal/cmd/api-key/command_delete.go

```diff
@@ -28,17 +25,44 @@ func (c *command) newDeleteCommand() *cobra.Command {
 
 func (c *command) delete(cmd *cobra.Command, args []string) error {
 	c.setKeyStoreIfNil()
-	apiKey := args[0]
 
-	promptMsg := fmt.Sprintf(errors.DeleteResourceConfirmYesNoMsg, resource.ApiKey, apiKey)
-	if ok, err := form.ConfirmDeletion(cmd, promptMsg, ""); err != nil || !ok {
+	if err := c.confirmDeletion(cmd, args); err != nil {
 		return err
 	}
-	httpResp, err := c.V2Client.DeleteApiKey(apiKey)
+
+	deleteFunc := func(id string) error {
+		if r, err := c.V2Client.DeleteApiKey(id); err != nil {
+			return errors.CatchApiKeyForbiddenAccessError(err, deleteOperation, r)
+		}
+		return nil
+	}
+
+	deleted, err := resource.Delete(args, deleteFunc, c.postProcess)
+	resource.PrintDeleteSuccessMsg(deleted, resource.ApiKey)
```

Comment:
Why are we printing a success message before we check for an error?

Reply:
`deleted` is a slice of resources which were successfully deleted. For each resource, `resource.Delete` either adds it's ID to `deleted` if there were no errors, or else adds the error to a multierror.

The idea here is that I print out the ones which succeeded, and then the error will tell the user which ones didn't delete.

Reply:
This seems a bit weird. How about we call `PrintDeleteSuccessMsg()` from inside `resource.Delete()`?